presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcSelectivePageSource.java

```diff
@@ -37,26 +40,41 @@
 public class OrcSelectivePageSource
         implements ConnectorPageSource
 {
+    static final int ROW_ID_COLUMN_INDEX = -10;
+
     private final OrcSelectiveRecordReader recordReader;
     private final OrcDataSource orcDataSource;
     private final OrcAggregatedMemoryContext systemMemoryContext;
     private final FileFormatDataSourceStats stats;
     private final RuntimeStats runtimeStats;
-
+    private final boolean appendRowNumberEnabled;
+    private final RowIDCoercer coercer;
+    private final boolean supplyRowIDs;
+    private final OptionalInt rowIDColumnIndex;
     private boolean closed;
 
-    public OrcSelectivePageSource(
+    OrcSelectivePageSource(
             OrcSelectiveRecordReader recordReader,
             OrcDataSource orcDataSource,
             OrcAggregatedMemoryContext systemMemoryContext,
             FileFormatDataSourceStats stats,
-            RuntimeStats runtimeStats)
+            RuntimeStats runtimeStats,
+            boolean appendRowNumberEnabled,
+            byte[] partitionID,
+            String rowGroupId,
+            boolean supplyRowIDs)
     {
         this.recordReader = requireNonNull(recordReader, "recordReader is null");
         this.orcDataSource = requireNonNull(orcDataSource, "orcDataSource is null");
         this.systemMemoryContext = requireNonNull(systemMemoryContext, "systemMemoryContext is null");
         this.stats = requireNonNull(stats, "stats is null");
         this.runtimeStats = runtimeStats;
+        this.appendRowNumberEnabled = appendRowNumberEnabled;
+        this.coercer = new RowIDCoercer(partitionID, rowGroupId);
+        // TODO can we combine supplyRowIDs and rowIDColumnIndex by using
+        // rowIDColumnIndex.isPresent() instead of a separate supplyRowIDs argument?
+        this.supplyRowIDs = supplyRowIDs;
+        this.rowIDColumnIndex = recordReader.toZeroBasedColumnIndex(ROW_ID_COLUMN_INDEX);
```

Comment:
Let's do the same assert that the rowIDColumnIndex is not empty if supplyRowIDs is true that we do in the factory